client/src/components/proposition/FormProposition.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from "react";
+import { TextField } from "@material-ui/core";
+import { Button } from "reactstrap";
+import { Redirect } from "react-router-dom";
+import axios from "axios";
+
+class FormProposition extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      code: "",
+      idProp: "",
+      redirect: false
+    };
+  }
+
+  componentDidMount() {
+    this.setState({
+      code: this.props.code,
+      idProp: this.props.idProp
+    });
+  }
+
+  handleChange = event => {
+    this.setState({ code: event.target.value });
+  };
+
+  onClickCompile = event => {
+    const idProp = this.state.idProp;
+    const source = this.state.code;
+
+    let base64 = require("base-64");
+    let username = this.props.currentUser.username;
+    axios
+      .get("http://localhost:55555/proposition/" + idProp + "/compile", {
+        params: {
+          source: source
+        },
+        headers: {
+          Authorization:
+            "Basic" +
```

Comment:
Does this work? doesn't need you a blank space after `Basic` to filter what is token from the word `Basic`???